osu.Game/Screens/Play/BreakOverlay.cs

```diff
@@ -134,51 +129,69 @@ public BreakOverlay(bool letterboxing, ScoreProcessor scoreProcessor)
         protected override void LoadComplete()
         {
             base.LoadComplete();
-            initializeBreaks();
 
             info.AccuracyDisplay.Current.BindTo(scoreProcessor.Accuracy);
             ((IBindable<ScoreRank>)info.GradeDisplay.Current).BindTo(scoreProcessor.Rank);
+
+            currentPeriod.BindTo(BreakTracker.CurrentPeriod);
+            currentPeriod.BindValueChanged(updateDisplay, true);
         }
 
+        private float remainingTimeForCurrentPeriod =>
+            currentPeriod.Value == null ? 0 : (float)Math.Max(0, (currentPeriod.Value.Value.End - Time.Current - BREAK_FADE_DURATION) / currentPeriod.Value.Value.Duration);
+
         protected override void Update()
         {
             base.Update();
 
             remainingTimeBox.Height = Math.Min(8, remainingTimeBox.DrawWidth);
         }
 
-        private void initializeBreaks()
+        protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint, EffectControlPoint effectPoint, ChannelAmplitudes amplitudes)
+        {
+            base.OnNewBeat(beatIndex, timingPoint, effectPoint, amplitudes);
+
+            if (currentPeriod.Value == null)
+                return;
+
+            float timeBoxTargetWidth = (float)Math.Max(0, (remainingTimeForCurrentPeriod - timingPoint.BeatLength / currentPeriod.Value.Value.Duration));
+            remainingTimeBox.ResizeWidthTo(timeBoxTargetWidth, timingPoint.BeatLength * 2, Easing.OutQuint);
```

Comment:
I also don't get why this transform has duration of *two* beat lengths? Won't that cause overlapping transforms if this fires on every beat?

Reply:
yeah it's just to make it really smooth (looks weird if it comes to a full stop).

Reply:
As cool as the beat-synced animation is, having played with it for a while now, I found that most of the time, it didn’t really fit the break of the songs I was playing. Since those sections are usually calmer with less things going on, the smooth animation just worked way better IMO.

I messed around with the multiplier next to `timingPoint.BeatLength` a bit. I tried 3, which worked okay. Then tried 4, that ended up being too smooth and made it feel delayed. Then landed on 3.5 as what I think would be ideal.

Reply:
Yeah I'm still not sold on it. We'll probably change the look of the overlay completely though. Feel free to submit a PR with your adjustment and I'll take it on board for consideration.